Hi. Again, my apologies that this has taken so long. I am getting the following error, when I run `make html`: 

```
D-108-179-172-65:doc (andim-apidocgeneration) $make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.3.4
making output directory...
Initializing GitHub plugin
loading pickled environment... not yet created
loading intersphinx inventory from http://docs.python.org/objects.inv...

Exception occurred:
  File "/Users/arokem/source/shablona/doc/tools/buildmodref.py", line 44, in writeapi
    source_version = __version__
NameError: name '__version__' is not defined
The full traceback has been saved in /var/folders/jz/4k438hl91v7gvdd592ct4lpc0000gn/T/sphinx-err-173mlzeu.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 1

```

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -16,6 +16,16 @@
 import sys
 import os
 
+# General information about the project.
+project = 'shablona'
+copyright = '2015, Ariel Rokem'
+
+currentdir = os.path.abspath(os.path.dirname(__file__))
+ver_file = os.path.join(currentdir, '..', project, 'version.py')
+with open(ver_file) as f:
+    exec(f.read())
+source_version = __version__
+
 currentdir = os.path.abspath(os.path.dirname(__file__))
 sys.path.append(os.path.join(currentdir, 'tools'))
 import buildmodref
@@ -24,8 +34,7 @@ import buildmodref
 # (see https://github.com/rtfd/readthedocs.org/issues/1139)
 def generateapidoc(_):
     output_path = os.path.join(currentdir, 'reference')
-    module = 'shablona'
-    buildmodref.writeapi(module, output_path, True)
+    buildmodref.writeapi(project, output_path, source_version, True)
 
 def setup(app):
     app.connect('builder-inited', generateapidoc)
@@ -70,9 +79,6 @@ source_suffix = '.rst'
 # The master toctree document.
 master_doc = 'index'
 
-# General information about the project.
-project = 'shablona'
-copyright = '2015, Ariel Rokem'
 
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
@@ -288,4 +294,3 @@ texinfo_domain_indices = False
 
 # Example configuration for intersphinx: refer to the Python standard library.
 intersphinx_mapping = {'http://docs.python.org/': None}
-

--- a/doc/tools/buildmodref.py
+++ b/doc/tools/buildmodref.py
@@ -6,7 +6,7 @@ from __future__ import print_function, division
 # stdlib imports
 import sys
 import re
-from os.path import join as pjoin
+import os.path
 
 # local imports
 from apigen import ApiDocWriter
@@ -38,7 +38,8 @@ def writeapi(package, outdir, other_defines=True):
 
     installed_version = V(module.__version__)
 
-    ver_file = pjoin('..', package, 'version.py')
+    currentdir = os.path.abspath(os.path.dirname(__file__))
+    ver_file = os.path.join(currentdir, '..', '..', package, 'version.py')
     with open(ver_file) as f:
         exec(f.read())
     source_version = __version__

--- a/doc/tools/buildmodref.py
+++ b/doc/tools/buildmodref.py
@@ -20,7 +20,8 @@ def abort(error):
     print('*WARNING* API documentation not generated: %s' % error)
     exit()
 
-def writeapi(package, outdir, other_defines=True):
+
+def writeapi(package, outdir, source_version, other_defines=True):
     # Check that the package is available. If not, the API documentation is not
     # (re)generated and existing API documentation sources will be used.
 
@@ -37,20 +38,12 @@ def writeapi(package, outdir, other_defines=True):
     # for older or newer versions if such versions are installed on the system.
 
     installed_version = V(module.__version__)
-
-    currentdir = os.path.abspath(os.path.dirname(__file__))
-    ver_file = os.path.join(currentdir, '..', '..', package, 'version.py')
-    with open(ver_file) as f:
-        exec(f.read())
-    source_version = __version__
-    print('***', source_version)
-
     if source_version != installed_version:
         abort("Installed version does not match source version")
 
     docwriter = ApiDocWriter(package, rst_extension='.rst',
                              other_defines=other_defines)
-                             
+
     docwriter.package_skip_patterns += [r'\.%s$' % package,
                                         r'.*test.*$',
                                         r'\.version.*$']

--- a/doc/tools/buildmodref.py
+++ b/doc/tools/buildmodref.py
@@ -6,7 +6,6 @@ from __future__ import print_function, division
 # stdlib imports
 import sys
 import re
-import os.path
 
 # local imports
 from apigen import ApiDocWriter

